xds/internal/xdsclient/xdsresource/unmarshal_rds.go

```diff
@@ -82,11 +83,41 @@ func unmarshalRouteConfigResource(r *anypb.Any, logger *grpclog.PrefixLogger) (s
 // we are looking for.
 func generateRDSUpdateFromRouteConfiguration(rc *v3routepb.RouteConfiguration, logger *grpclog.PrefixLogger, v2 bool) (RouteConfigUpdate, error) {
 	vhs := make([]*VirtualHost, 0, len(rc.GetVirtualHosts()))
+	csps := make(map[string]clusterspecifier.BalancerConfig)
```

Comment:
Can this block of code be moved into its own function?

Reply:
Yeah, good idea. Moved to own function.